### PR TITLE
"waypoint user" CLI

### PR DESCRIPTION
**Pull request by mitchellh**
This introduces a "waypoint user" CLI that is more user-oriented. This deprecates `waypoint token`. I kept the `waypoint token` command around so it works still but it is now hidden.

This exposes the user system introduced in #1752 and available as part of #1831.

### CODE REVIEW

**Review by izaaklauer — APPROVED** [review]
All looks good to me. I can imagine operators wanting a `user list` command, but it doesn't need to come now.

File: internal/cli/main.go [hunk 1]
```diff
@@ -48,6 +48,9 @@ var (
 	// hiddenCommands are not shown in CLI help output.
 	hiddenCommands = map[string]struct{}{
 		"plugin": {},
+
+		// Deprecated:
+		"token": {}, // replaced by "user"
```
[the remark this answers is not in the document]
Comment (briancain) [reply internal/cli/main.go:53]: Agreed, there are also a handful of places on the website that reference `waypoint token new`:
```
website/content/docs/automating-execution/index.mdx
43:- `WAYPOINT_SERVER_TOKEN`. Must be set to a Waypoint token, created with [`waypoint token new`](/commands/token-new)

website/content/docs/runner/run-manual.mdx
54:  auth token using `waypoint token new`.

website/content/docs/server/auth.mdx
54:$ waypoint token new

website/content/docs/troubleshooting.mdx
175:$ waypoint token new

website/content/commands/token-new.mdx
8:# Waypoint Token new
10:Command: `waypoint token new`
18:Usage: `waypoint token new [options]`

ui/app/components/login-token/index.hbs
5:      <pre><code id="waypoint-token-new">waypoint token new</code></pre>
```

**Review by briancain — CHANGES_REQUESTED** [review]
Looks good!

If we're deprecating token we should add a warning message in that command too that prints right away when invoked.
The website and UI updates could come in a separate PR though.

**Comment by mitchellh** [comment]
> If we're deprecating token we should add a warning message in that command too that prints right away when invoked.

I was kind of afraid to do this cause right now we ONLY output the token so its easy to pipe somewhere. Maybe I'll force send it to stderr that should fix that.

**Comment by mitchellh** [comment]
I'll merge this and do the docs changes as part of the user system docs in general. There's a lot there.

**Review by briancain — APPROVED** [review]

**Comment by fr3fou** [comment]
image
The new set of commands don't seem to be working for me (running this on my server)

**Comment by briancain** [comment]
Hey there @fr3fou - could you please open a new issue? We'll be able to help triage what's going on there better, thanks!

**Pull request merged**